sources/OpenMcdf.Extensions/OLEProperties/OLEPropertiesContainer.cs

```diff
@@ -175,6 +175,44 @@ public void AddProperty(OLEProperty property)
             properties.Add(property);
         }
 
+        /// <summary>
+        /// Create a new UserDefinedProperty.
+        /// </summary>
+        /// <param name="vtPropertyType">The type of property to create.</param>
+        /// <param name="name">The name of the new property.</param>
+        /// <returns>The new property, of null if this container can't contain user defined properties.</returns>
+        public OLEProperty AddUserDefinedProperty(VTPropertyType vtPropertyType, string name)
+        {
+            if (this.ContainerType != ContainerType.UserDefinedProperties)
```

Comment:
Maybe it could be done so that calls against a DocumentSummaryInfo container forward on to the contained UserDefinedProperties container. (Now that there is a CreateUserDefinedProperties() function, it could even create the UserDefinedProperties container if there isn't one already)